SingularityUI/app/components/webhooks/Webhooks.cjsx

```diff
@@ -0,0 +1,215 @@
+            },
+            {
+                data: 'Queue Size'
+                sortable: true
+                doSort: (sortDirectionAscending) => sortBy 'queueSize', sortDirectionAscending
+            },
+            {
+                className: 'hidden-xs'
+            }
+        ]
+
+    deleteWebhook: (webhook) ->
+        $.ajax
+            url: "#{ config.apiRoot }/webhooks/?webhookId=#{webhook.attributes.webhook.id}"
+            type: "DELETE"
+            success: () => @props.collections.webhooks.fetch().done => @forceUpdate()
+
+    promptDeleteWebhook: (webhook) ->
+        deleteWebhook = (webhook) => @deleteWebhook webhook
+        vex.dialog.confirm
+            message: "<div class='delete-webhook' />" # This is not react
+            afterOpen: =>
+                ReactDOM.render(
+                    <div>
+                        <pre>({webhook.attributes.webhook.type}) {webhook.attributes.webhook.uri}</pre>
+                        <p>Are you sure you want to delete this webhook?</p>
+                    </div>,
+                    $(".delete-webhook").get(0)
+                )
+            callback: (confirmed) =>
+                return unless confirmed
+                deleteWebhook webhook
+
+    newWebhook: (uri, type) ->
+        data = {
+            uri: uri
+            type: type
+        }
+        data.user = app.user.attributes.user.id if app.user.attributes.authenticated
+        $.ajax
+            url: "#{ config.apiRoot }/webhooks"
+            type: "POST"
+            contentType: 'application/json'
+            data: JSON.stringify data
+            success: () => @props.collections.webhooks.fetch().done => @forceUpdate()
+
+    promptNewWebhook: ->
+        newWebhook = (uri, type) => @newWebhook uri, type
+        vex.dialog.open
+            message: "<div class='new-webhook' />"
+            afterOpen: =>
+                @validateInput = (input) =>
+                    try
+                        new URL input
+                        return true
+                    catch err
+                        return false
+                @renderedForm = ReactDOM.render(
+                    <NewWebhookForm
+                        getErrors = {() => @errors}
+                        webhookTypes = {@webhookTypes}
+                        setType = {(selected) => @type = selected} 
+                        setUri = {(uri) => @uri = uri} />,
+                    $(".new-webhook").get(0)
+                )
+            beforeClose: =>
+                return true unless @data
+                @errors = []
+                uriValidated = @validateInput @uri
+                @errors.push 'Please select a type' unless @type
+                @errors.push 'Invalid URL entered' unless uriValidated
+                @renderedForm.forceUpdate() unless uriValidated and @type
+                return false unless uriValidated
+                return false unless @type
```

Comment:
I looked over the code and went through the UI, but I couldn't find a default. If you don't select a type the request isn't sent, the dialog stays open, and the UI prompts you to select a type.
On a previous iteration I had a default, but I removed that. It's possible that there are artifacts left over from that somewhere, but I can't find them at the moment. Where are you seeing a default? I'll remove it if I can find it.